corehq/apps/locations/views.py

```diff
@@ -252,10 +252,12 @@ def mk_loctype(name, parent_type, administrative,
             pk = loc_type['pk']
             if not _is_fake_pk(pk):
                 pks.append(loc_type['pk'])
-        if not (
-            len(loc_types) == len(set(lt['name'] for lt in loc_types))
-            and len(loc_types) == len(set(lt['code'] for lt in loc_types))
-        ):
+
+        names = [lt['name'] for lt in loc_types]
+        unique_names = len(names) == len(set(names))
+        codes = [lt['code'] for lt in loc_types if lt['code']]
+        unique_codes = len(codes) == len(set(codes))
+        if not unique_names or not unique_codes:
```

Comment:
boolean names might be clearer here - `unique_codes` sounds like a list of unique codes.  Something like:
```python
has_duplicate_codes = len(codes) != len(set(codes))
```